src/kubernetes-ingress-controller/guides/upgrade-kong-3x.md

```diff
@@ -79,18 +79,25 @@ prefix.
 Ingress rule paths have no way to indicate that a path is a regular expression.
 The `ImplementationSpecific` path type can contain either regular expression or
 non-regular expression paths. If you have existing Ingresses with regular
-expression paths, those paths will no longer be routed correctly if you upgrade
+expression paths, those paths will break if you upgrade
 to 3.x without updating configuration.
```

Comment:
Slotting the additional warning from #4471 in here arbitrarily. It's not directly related to any upgrade concept, but I want to remind about this because we see this misconfiguration occasionally, and it'll cause worse problems with 3.x.

```suggestion
expression paths, those paths will break if you upgrade
to 3.x without updating configuration.

{:.important}
> **Important**: `Prefix` and `Exact` rules [must never use regular expressions](/kubernetes-ingress-controller/{{page.kong_version}}/concepts/ingress-versions/#networkingk8siov1).
Only use regular expressions in `ImplementationSpecific` rules.
```